### DISCUSSION TITLE
Help needed: TypeError when filtering data in server load function

### DISCUSSION THREAD

**Discussion by ABarLT** [huggingface/chat-ui#1412] [Q&A] [answered]
I have a fork of the repo and when I am running it, I'm encountering a TypeError in the load function of the src/routes/+layout.server.ts file. The specific error message is:

TypeError: ((settings?.tools ?? []).filter is not a function

This happens on line 114 of the src/routes/+layout.server.ts file. It seems that the filter method is being called on a value that is either null, undefined, or not an array.

The error occurs when trying to load the '/chat/' route on my local development server.

Can anyone help me understand why this error might be occurring and how to resolve it?

**Comment by nsarrazin** [comment, marked as answer]
Oh I think I see what the issue might be. We switched the `settings` from an object to an array and had a migration for it but it only runs for huggingchat. https://github.com/huggingface/chat-ui/blob/eee852561ae25ab6c356df901b78fc111c90589c/src/lib/migrations/routines/07-reset-tools-in-settings.ts#L16

Will submit a fix

**Reply by nsarrazin** [reply]
Ok the changes have been added, try merging our latest main into your fork to see if that improves things

**Reply by ABarLT** [reply]
Yes, that fixed it! Thank you!
